### ISSUE TITLE
xml parse functionality: missing start or close tag on xml document doesn't report error if on the same line

### ISSUE THREAD

**Issue by kaizer200** [prettydiff/prettydiff#204]
**_Example**_

``` xml
<?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE david SYSTEM "Note.dtd">
 <tag1></tag2>
```

**_expected result**_
problem reported on the third line, stating that tag1 closing tag is missing, or tag2 starting tag is missing
**_result**_
no error issued
**_addition**_
if tag1 and /tag2 were on different lines, the error was issued

**Comment by prettydiff** [comment]
The code example is indeed non-valid XML.  At this time Pretty Diff is not performing code analysis or validation.  Pretty Diff will parse this correctly according to the syntax and improperly indicate the `</tag2>` is the closing tag to `<tag1>`.  Pretty Diff is not performing any sort of grammar detection except where certain template languages have explicit grammars that doubles as parsing syntax.

In summary the provide code sample is a valid defect and Pretty Diff is not fixing this in version 3.
